src/Bridges/Psr/Cache.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * This file is part of the Tracy (https://tracy.nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+declare(strict_types=1);
+
+namespace Nette\Bridges\Psr;
+
+use DateInterval;
+use Generator;
+use Nette;
+use Psr;
+
+class Cache implements Psr\SimpleCache\CacheInterface
+{
+	public function __construct(private Nette\Caching\Storage $storage)
+	{
+	}
+
+
+	public function get(string $key, mixed $default = null): mixed
+	{
+		return $this->storage->read($key) ?? $default;
+	}
+
+
+	public function set(string $key, mixed $value, null|int|DateInterval $ttl = null): bool
+	{
+		$dependencies = [];
+		if ($ttl !== null) {
+			$dependencies[Nette\Caching\Cache::Expire] = self::ttlToSeconds($ttl);
+		}
+
+		$this->storage->write($key, $value, $dependencies);
+
+		return true;
+	}
+
+
+	public function delete(string $key): bool
+	{
+		$this->storage->remove($key);
+
+		return true;
+	}
+
+
+	public function clear(): bool
+	{
+		$this->storage->clean([Nette\Caching\Cache::All => true]);
+
+		return true;
+	}
+
+
+	/**
+	 * @return Generator<string, mixed>
+	 */
+	public function getMultiple(iterable $keys, mixed $default = null): iterable
+	{
+		foreach ($keys as $name) {
+			yield $name => $this->get($name, $default);
+		}
+	}
+
+
+	/**
+	 * @param iterable<string|int|float, mixed> $values
+	 */
+	public function setMultiple(iterable $values, null|int|DateInterval $ttl = null): bool
+	{
+		$ttl = self::ttlToSeconds($ttl);
+
+		foreach ($values as $key => $value) {
+			$this->set((string) $key, $value, $ttl);
+		}
+
+		return true;
+	}
+
+
+	public function deleteMultiple(iterable $keys): bool
+	{
+		foreach ($keys as $value) {
+			$this->delete($value);
+		}
+
+		return true;
+	}
+
+
+	public function has(string $key): bool
+	{
+		return $this->storage->read($key) !== null;
+	}
+
+	private static function ttlToSeconds(null|int|DateInterval $ttl = null): ?int
+	{
+		if ($ttl instanceof DateInterval) {
+			return self::dateIntervalToSeconds($ttl);
+		}
+
+		return $ttl;
+	}
+
+	private static function dateIntervalToSeconds(DateInterval $dateInterval): int
+	{
+		if ($dateInterval->days === false) {
+			$days = $dateInterval->y * Nette\Utils\DateTime::YEAR
+				+ $dateInterval->m * Nette\Utils\DateTime::MONTH
+				+ $dateInterval->d * Nette\Utils\DateTime::DAY;
+		} else {
+			$days = $dateInterval->days * Nette\Utils\DateTime::DAY;
+		}
+
+		$days += $dateInterval->h * Nette\Utils\DateTime::HOUR
+			+ $dateInterval->i * Nette\Utils\DateTime::MINUTE
+			+ $dateInterval->s;
+
+		return $dateInterval->invert === 1 ? $days * -1 : $days;
```

Comment:
Sth like this should be less error-prone.

```suggestion
		$now = new DateTimeImmutable();
		$expiresAt = $now->add($dateInterval);
		return $expiresAt->getTimestamp() - $now->getTimestamp();
```

Reply:
In my case, the result is same like in tests
```php
$from = new DateTime('1978-01-23 05:06:07');
$to = new DateTime('1986-12-30 07:08:09');
Assert::same(281930522, $to->getTimestamp() - $from->getTimestamp());
```

In your case is the result is `282016922`.

I have to think about what is right.

Reply:
You can see in last commit de582d0e1dbd5eb9110b790ef3d431b33613b438.

Reply:
I got it, your solution is better, for human understanding of time.

```
input unix time: 1716720092 -> 2024-05-26 12:41:32

$interval = new DateInterval('P3Y6M4DT12H30M5S');

'2024-05-26 12:41:32' + 5s + 30m + 12h + 4d + 6m + 3y = 2027-12-01 01:11:37 
// 110899805 -> 2027-12-01 01:11:37

1716720092 + 110842205 = 2027-11-30 09:11:37
// 110842205 -> 2027-11-30 09:11:37
```